hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/MapLongByteArrayCodec.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.codec.builtin;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+
+import static com.hazelcast.client.impl.protocol.ClientMessage.BEGIN_FRAME;
+import static com.hazelcast.client.impl.protocol.ClientMessage.END_FRAME;
+
+public final class MapLongByteArrayCodec {
+
+    private MapLongByteArrayCodec() {
+    }
+
+    public static void encode(ClientMessage clientMessage, Collection<Map.Entry<Long, byte[]>> collection) {
+        List<Long> valueList = new ArrayList<>(collection.size());
+        clientMessage.add(BEGIN_FRAME);
+        for (Map.Entry<Long, byte[]> entry : collection) {
+            valueList.add(entry.getKey());
+            ByteArrayCodec.encode(clientMessage, entry.getValue());
+        }
+        clientMessage.add(END_FRAME);
+
+        ListLongCodec.encode(clientMessage, valueList);
+    }
+
+    public static List<Map.Entry<Long, byte[]>> decode(ListIterator<ClientMessage.Frame> iterator) {
```

Comment:
why not return Map? The name of the codec is misleading.

Reply:
I used the same naming convention, we can rename them if we want so let's keep it like other builtin codecs